src/main/java/com/commercetools/sync/services/impl/UnresolvedReferencesServiceImpl.java

```diff
@@ -60,6 +63,13 @@ public UnresolvedReferencesServiceImpl(@Nonnull final ProductSyncOptions baseSyn
     @Nonnull
     @Override
     public CompletionStage<Optional<WaitingToBeResolved>> save(@Nonnull final WaitingToBeResolved draft) {
+
+        if (isBlank(draft.getProductDraft().getKey())) {
+            syncOptions.applyErrorCallback(format(SAVE_FAILED, draft.getProductDraft().getKey(),
+                    "Draft key is blank!"));
```

Comment:
1. It's nice to have but we don't need it at the moment (https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) since we know we always call it with a non-blank key. So there is no need to add the logic and test it and so on. 
2. "Draft key is blank!" is not very helpful for the user in this case, some information in the draft like the value maybe could be useful for the user to know more information about this case.

Reply:
1e2a4526c537a0e4abe2a909bfa0b007a23ec7f5